grpc-gateway/pb/devices.go

```diff
@@ -64,3 +69,28 @@ func (e *Event_ResourceCreated) GetResourceId() *commands.ResourceId {
 	}
 	return e.ResourceCreated.GetResourceId()
 }
+
+func (f *ResourceIdFilter) ToString() string {
+	if f == nil {
+		return ""
+	}
+	var sb strings.Builder
+	sb.WriteString(f.GetResourceId().GetDeviceId())
+	if f.GetResourceId().GetHref() == "" {
+		return sb.String()
+	}
+	sb.WriteString(f.GetResourceId().GetHref())
+	if len(f.GetEtag()) == 0 {
+		return sb.String()
+	}
+	for i, etag := range f.GetEtag() {
+		if i == 0 {
+			sb.WriteString("?")
+		} else {
+			sb.WriteString("&")
+		}
+		sb.WriteString("etag=")
+		sb.WriteString(base64.StdEncoding.EncodeToString(etag))
```

Comment:
Shouldn't `base64.RawURLEncoding` be used to avoid the use of URL problematic characters? If it's never used in an URI then it's OK.